eng/tools/select-packages/index.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See the @microsoft/rush package's LICENSE file for license information.
+
+// This script searches for package.json files recursively under a given path
+// excluding packages under a node_modules folder and opens them to determine
+// whether they contain a "sdk-archetype" field with the value of "client".
+//
+// If they do, then the package name is recorded and then after all package.json files
+// have been scanned a single string is output using the Azure Pipelines Commands special
+// syntax which creates a variable for use within a later stage of the pipeine.
+var glob = require("glob");
+
+let logLineCounter = 0;
+const log = message => {
+  console.log(`select-packages(${logLineCounter++}): ${message}`);
+};
+
+log(`Working directory is "${process.cwd()}".`);
+
+let filter = process.argv[2];
+log(`Specified service directory is "${filter}".`);
+
+// This code assumes that in the service directory, packages that we want to publish
+// exist only in directories that are immediate children of the service directory.
+glob(filter, (err, files) => {
+  if (err) {
+    log(`An error occured: ${err}`);
+    process.exit(1);
+  }
+
+  let packageTargets = "";
+
+  if (files) {
+    log(`Found ${files.length} packages under service directory.`);
+
+    for (let fileIndex in files) {
+      log(`Processing package file ${fileIndex} at "${files[fileIndex]}"`);
+      let packageContents = require(files[fileIndex]);
+
+      log(
+        `Package name for package ${fileIndex} is "${packageContents.name}".`
+      );
+
+      if (packageContents["sdk-type"] == "client") {
```

Comment:
Perhaps not now but we will likely want to pass in the sdk-type as an argument to this script when we decide we want to scope a job to client or data only.

Reply:
I've got some grander plans here to change this command to support something like:

```
$ select-packages --glob sdk/**/!(node_modules)/package.json --query [jmespath] 
```

But didn't want to overbake it. If I go further with it then I'd probably create a separate open source tool for doing this kind of thing :P

Reply:
There's an existing `rush list-files` command that will dump the file list of all projects in the Rush workspace (it's just a script in common/scripts so you should be able to import it too). Not sure if that's helpful here or not but you could just loop through that output for package.json files, and then do your query from there.